File: integrations/aws-strands/typescript/examples/server/api/agentic-chat-reasoning.ts
/**
 * Agentic Chat with Reasoning example for AWS Strands (TypeScript).
 *
 * Demonstrates reasoning/thinking event streaming. When the underlying model
 * supports extended thinking, the adapter emits REASONING_* events that the
 * frontend can display as a "thinking" indicator.
 */

import { Agent } from "@strands-agents/sdk";
import { StrandsAgent } from "@ag-ui/aws-strands";
import { createStrandsApp } from "@ag-ui/aws-strands/server";
import { createModel } from "../model-factory";
import { demoPort, listenOrExit, runIfMain } from "../run-if-main";

export const SYSTEM_PROMPT = `
    You are a helpful assistant that thinks through problems step by step.
    When the user greets you, always greet them back. Your greeting should always start with "Hello".
    Your greeting should also always ask (exact wording) "how can I assist you?"
    When reasoning about a problem, break it down into clear steps before answering.
  `;

export async function createAgenticChatReasoningAgent(): Promise<StrandsAgent> {
  return new StrandsAgent({
    agent: new Agent({
      // API mode named rather than inherited: reasoning summaries are what
      // this demo exists to show, and they only come back on the Responses
      // API. The Python reference names it here for the same reason.
      model: await createModel({ openaiApi: "responses", reasoning: true }),
      systemPrompt: SYSTEM_PROMPT,
    }),
    name: "agentic_chat_reasoning",
    description:
      "Conversational Strands agent with reasoning/thinking event streaming",
  });
}

runIfMain(import.meta.url, async () => {
  // Port first: it throws on a malformed PORT, and building the agent first
  // would surface a missing API key instead and hide the real complaint.
  const port = demoPort();
  const app = await createStrandsApp(await createAgenticChatReasoningAgent(), {
    path: "/",
  });
  listenOrExit(app, "agentic-chat-reasoning", port);
});

File: integrations/aws-strands/python/examples/server/api/agentic_chat_reasoning.py
"""Agentic Chat with Reasoning example for AWS Strands.

Demonstrates reasoning/thinking event streaming. When the underlying model
supports extended thinking, the adapter emits REASONING_* events that the
frontend can display as a "thinking" indicator.
"""
import os
from pathlib import Path
from dotenv import load_dotenv

# Suppress OpenTelemetry context warnings
os.environ["OTEL_SDK_DISABLED"] = "true"
os.environ["OTEL_PYTHON_DISABLED_INSTRUMENTATIONS"] = "all"

from strands import Agent
from ag_ui_strands import StrandsAgent, create_strands_app
from server.model_factory import create_model
from server.settings import cors_origins

# Load environment variables from .env file
env_path = Path(__file__).parent.parent.parent / '.env'

load_dotenv(dotenv_path=env_path)

# Create model from MODEL_PROVIDER env var (default: openai).
# This demo is specifically about reasoning, so it opts into the OpenAI
# Responses API, which is what surfaces reasoning summaries. Every other
# demo uses the factory default (Chat Completions, no reasoning).
model = create_model(openai_api="responses", reasoning=True)

strands_agent = Agent(
    model=model,
    system_prompt="""
    You are a helpful assistant that thinks through problems step by step.
    When the user greets you, always greet them back. Your greeting should always start with "Hello".
    Your greeting should also always ask (exact wording) "how can I assist you?"
    When reasoning about a problem, break it down into clear steps before answering.
    """,
)

agui_agent = StrandsAgent(
    agent=strands_agent,
    name="agentic_chat_reasoning",
    description="Conversational Strands agent with reasoning/thinking event streaming",
)

app = create_strands_app(agui_agent, "/", origins=cors_origins())
